### ISSUE TITLE
A question about this action

### ISSUE THREAD

**Issue by shanebenlolo** [Sibz/github-status-action#28]
Hello, 

I am a little confused about what this action does. Does it explicitly set the status to the value provided in the `state` parameter? Sorry for the confusion, any help would be appreciated thank you.

**Comment by shanebenlolo** [comment]
I now understand what this action is doing. Sorry the confusion.

**Comment by kasir-barati** [comment]
@shanebenlolo thanks a lot for bringing up the question, tbf I am also not sure what this actions is doing for us and I really appreciate it if you could shed some light on this topic since it is not quite clear to me just by reading the README.md.

@Sibz could you please add more info so that newbies like me do not have to spend too much time just to understand what it does?

Many thanks

**Comment by shanebenlolo** [comment]
> @shanebenlolo thanks a lot for bringing up the question, tbf I am also not sure what this action is doing for us and I really appreciate it if you could shed some light on this topic since it is not quite clear to me just by reading the README.md.
> 
> @Sibz could you please add more info so that newbies like me do not have to spend too much time just to understand what it does?
> 
> Many thanks

Hey unfortunately I cannot recall what this was used for as it was many years ago. I should have documented my findings, sorry about that. Best of luck.

**Comment by kasir-barati** [comment]
Found it @shanebenlolo, it is useful in scenarios that you wanna specify the state of a commit. It's easier to explain it with this example;
1. I wanna run my e2e tests, so first I need to specify the state of the commit (last commit that we pushed, assuming the action gets executed on push) to `pending`.
2. Then I run e2e tests
3. Now I wanna change the state of the commit to `success` or `failure` or `error` based on the second step

Now it is documented :grin:
